### ISSUE TITLE
add support to copy files between two nodes in `tsh scp`

### ISSUE THREAD

**Issue by tukusejssirs** [gravitational/teleport#23819]
## What would you like Teleport to do?

I would like to simply copy files between two nodes, like `tsh ssh user1@host1:/file user2@host2:/`. Of course, local machine or Teleport cluster needs to serve as proxy.

## What problem does this solve?

Currently, we need to accomplish this in two separate commands.

In the end, I think `tsh ssh` should accomplish feature parity with `scp` from `openssh`. Or even better: accomplish feature parity with `rsync` (or use it instead in the background).

## If a workaround exists, please include it.

Copy the file in two steps:
1. Copy it to local machine:

```bash
tsh ssh user1@host1:/file1 ./
```

2. Then upload it to remote server:

```bash
tsh ssh ./ user2@host2:/
```

**Comment by stevenGravy** [comment]
headless sso should allow this since you'll be to authenticate remotely on a node.

**Comment by tukusejssirs** [comment]
> headless sso should allow this since you'll be to authenticate remotely on a node.

I am not sure if I understand you correctly.

Currently, it does not work for me:

```bash
tsh scp user1@host1:/file1 user2@host2:/

# Output
ERROR: could not access local path "user1@host1:/file1": stat user1@host1:/file1: no such file or directory

ERROR: could not access local path "user1@host1:/file1": stat user1@host1:/file1: no such file or directory
```

File `user1@host1:/file1` exists on that host (I have created it just before the test).

Note, however, that the two nodes are in different locations/servers/LANs.

**Comment by stevenGravy** [comment]
Yes I don't expect we'll support that soon. We are heading in headless sso so you could be on a node and then copy from that node to another.

**Comment by tukusejssirs** [comment]
Okay, logging into a node (e.g. `tsh ssh user1@host1`, running `tsh scp file user2@host2 --user tp_user --proxy teleport.example.com` copies the file as expected. Therefore I can run `tsh ssh -t user1@host1 'tsh scp file user2@host2:/ --proxy teleport.example.com --user tp_user` to do what.

However, you said that I’ll _be to authenticate remotely on a node_. I’ve thought it would authenticate me automatically, but it is not the case. Could Teleport authenticate me automatically when I log in using `tsh login` on my local machine and connect to a node? Of course, I need to have access to both nodes.

**Comment by zmb3** [comment]
Looks like a duplicate of #7915

**Comment by tukusejssirs** [comment]
Right, thanks @zmb3! :pray: While I search the issues, I did not find that issue.
